sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -428,6 +441,25 @@ protected ApplicationGatewayBackendImpl ensureUniqueBackend() {
         return backend;
     }
 
+    private void ensureNoMixedWaf() {
+        String errorMessage = "Can't have a mixture of legacy WAF configuration and WAF policy. "
+            + "If you are using legacy WAF configuration, you are strongly encouraged to upgrade to WAF Policy "
+            + "for easier management, better scale, and a richer feature set at no additional cost. "
+            + "See https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/upgrade-ag-waf-policy";
+        if (this.creatableWafPolicy != null || this.innerModel().firewallPolicy() != null) {
+            if (isInCreateMode()) {
+                if (this.legacyWafConfigurationSpecified) {
+                    throw new IllegalStateException(errorMessage);
```

Comment:
Should this check be earlier?

In `beforeCreating()`, I assume `creatableWafPolicy` could be already created (since you use its id there)?

Reply:
Yes. It should happen before any dependency task run. I'll put the check in `beforeGroupCreateOrUpdate`. Thanks for the catch!

Reply:
Changed, thanks. Also added a test scenario.